frontend/src/concepts/pipelines/content/artifacts/ArtifactUriLink.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import { useIsAreaAvailable, SupportedArea } from '~/concepts/areas';
+import { extractS3UriComponents } from './utils';
+
+interface ArtifactUriLinkProps {
+  uri?: string;
+}
+
+export const ArtifactUriLink: React.FC<ArtifactUriLinkProps> = ({ uri }) => {
+  const { namespace } = usePipelinesAPI();
+  const isS3EndpointAvailable = useIsAreaAvailable(SupportedArea.S3_ENDPOINT).status;
+
+  const url = React.useMemo(() => {
+    if (!uri || !isS3EndpointAvailable) {
+      return;
+    }
+
+    // Check if the uri starts with http or https return it as is
+    if (uri.startsWith('http://') || uri.startsWith('https://')) {
+      return uri;
+    }
+
+    // Otherwise check if the uri is s3
+    // If it is not s3, return undefined as we only support fetching from s3 buckets
+    const uriComponents = extractS3UriComponents(uri);
+    if (!uriComponents) {
+      return;
+    }
+
+    const { bucket, path } = uriComponents;
+
+    // /api/storage/${namespace}/${bucket}?key=${path}
```

Comment:
Not sure what this comment is telling that the code immediately under it doesn't show almost directly.